File: src/main/java/net/azisaba/rccore/integration/auctionhouse/MarketEventListener.java
package net.azisaba.rccore.integration.auctionhouse;

import com.spawnchunk.auctionhouse.events.ListItemEvent;
import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;

import java.util.List;

public class MarketEventListener implements Listener {
    // TODO: 販売禁止アイテムに関する実装
    @EventHandler
    public void onListNewItem(ListItemEvent event) {
        List<Component> loreList = event.getItem().lore();
        if(loreList == null) return;
        loreList.stream()
                .map(l -> LegacyComponentSerializer.legacyAmpersand().serialize(l))
                .anyMatch(l -> l.contains(""));
    }
}

File: src/main/kotlin/net/azisaba/rccore/integration/auctionhouse/MarketEventListener.kt
package net.azisaba.rccore.integration.auctionhouse

import com.badbones69.crazyauctions.api.events.AuctionListEvent
import net.azisaba.rccore.config
import net.kyori.adventure.text.Component
import net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener

class MarketEventListener : Listener {
    @EventHandler
    fun onListNewItem(event: AuctionListEvent) {
        val loreList = event.item.lore()
        if (loreList == null) return

        val isDenyItem = loreList.stream()
            .map { l: Component -> LegacyComponentSerializer.legacyAmpersand().serialize(l) }
            .anyMatch { l: String -> config.integrationSetting.marketDenyItemLore.stream().anyMatch { s: String -> l.contains(s) } }

        if (isDenyItem) {
            event.isCancelled = true
            // check: 何かしらのプレイヤーへのフィードバックがいるかも？
        }
    }
}
